src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/13.sql

```diff
@@ -0,0 +1,3443 @@
+    @queueId varchar(64),
+	@count smallint,
+    @taskHeartbeatTimeoutThresholdInSeconds int = 600
+AS
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    -- We will consider a job to be stale if its timestamp is smaller than or equal to this.
+    DECLARE @expirationDateTime dateTime2(7)
+    SELECT @expirationDateTime = DATEADD(second, -@taskHeartbeatTimeoutThresholdInSeconds, SYSUTCDATETIME())
+
+    DECLARE @availableJobs TABLE (
+		TaskId varchar(64),
+		QueueId varchar(64),
+		Status smallint,
+        TaskTypeId smallint,
+		IsCanceled bit,
+        RetryCount smallint,
+		HeartbeatDateTime datetime2,
+		InputData varchar(max),
+		TaskContext varchar(max),
+        Result varchar(max)
+	)
+
+    INSERT INTO @availableJobs
+    SELECT TOP(@count) TaskId, QueueId, Status, TaskTypeId, IsCanceled, RetryCount, HeartbeatDateTime, InputData, TaskContext, Result
+    FROM dbo.TaskInfo
+    WHERE (QueueId = @queueId AND ((Status = 1 OR (Status = 2 AND HeartbeatDateTime <= @expirationDateTime)) AND IsCanceled = 0))
+    ORDER BY HeartbeatDateTime
+
+    DECLARE @heartbeatDateTime datetime2(7) = SYSUTCDATETIME()
+
+    UPDATE dbo.TaskInfo
+    SET Status = 2, HeartbeatDateTime = @heartbeatDateTime, RunId = CAST(NEWID() AS NVARCHAR(50))
+    FROM dbo.TaskInfo task INNER JOIN @availableJobs availableJob ON task.TaskId = availableJob.TaskId
+
+	Select task.TaskId, task.QueueId, task.Status, task.TaskTypeId, task.RunId, task.IsCanceled, task.RetryCount, task.MaxRetryCount, task.HeartbeatDateTime, task.InputData, task.TaskContext, task.Result
+	from dbo.TaskInfo task INNER JOIN @availableJobs availableJob ON task.TaskId = availableJob.TaskId
+
+    COMMIT TRANSACTION
+GO
+
+/*************************************************************
+    Capturing changes for event
+**************************************************************/
+
+/*************************************************************
+    Resource change data table
+**************************************************************/
+
+CREATE TABLE dbo.ResourceChangeData
+(
+    Id bigint IDENTITY(1,1) NOT NULL,
+    Timestamp datetime2(7) NOT NULL,
+    ResourceId varchar(64) NOT NULL,
+    ResourceTypeId smallint NOT NULL,
+    ResourceVersion int NOT NULL,
+    ResourceChangeTypeId smallint NOT NULL
+PRIMARY KEY CLUSTERED 
+(
+    [Id] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+ALTER TABLE dbo.ResourceChangeData ADD  CONSTRAINT DF_ResourceChangeData_Timestamp  DEFAULT (sysutcdatetime()) FOR Timestamp
+GO
+
+/*************************************************************
+    Resource change type table
+**************************************************************/
+CREATE TABLE dbo.ResourceChangeType(
+	ResourceChangeTypeId smallint IDENTITY(0,1) NOT NULL,
```

Comment:
Nit: what's the convention for indention?

Reply:
It is 4 spaces.